pygmt/tests/test_meca.py

```diff
@@ -0,0 +1,156 @@
+"""
+Tests for meca
+"""
+import os
+
+import pytest
+
+from .. import Figure
+
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+
+
+@pytest.mark.mpl_image_compare
+def test_meca_full_moment_tensor():
+    """
+    Check full moment tensor, especially with big isotropic components. The
+    data files and original script are provided by Carl Tape, available from
+    https://github.com/carltape/compearth, licensed under MIT license. Also see
+    https://github.com/GenericMappingTools/gmt/issues/661.
+    """
+
+    fig = Figure()
+
+    for file in (
+        "fullmt_ipts1_iref1",
```

Comment:
By the way, I don't know if this `meca` implementation works with `pandas.DataFrame`s yet, but just building on from DongDong's answer at https://forum.generic-mapping-tools.org/t/accessing-gmt-test-suite-files-using-syntax/655/3, you can actually 'download' the data files directly using [`pandas.read_csv`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html)  like so:

```python
df = pd.read_csv("https://raw.githubusercontent.com/GenericMappingTools/gmt/master/test/seis/fullmt_ipts1_iref1", sep="\s+", header=None)
```

And maybe use [`pd.to_csv`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html) to output to a file. Or you could wait for it to show up on the GMT cache.

Reply:
We can add these files to the GMT cache, but I don't think we should.

GMT already has many tests checking if meca works well with various focal mechanisms. This PyGMT test is doing the same thing, and it's testing GMT, not PyGMT. 

IMHO, the PyGMT tests should focus on testing the Python codes, for examples, there are several different ways to pass focal mechanisms to `meca()` to test.

- pass a file
- passing an array 
- pass a single focal mechanism like `dict(strike=0, dip=90, rake=0, magnitude=5),`
- pass multiple focal mechanisms like `dict(strike=[0, 30], dip=[30, 60], rake=[30, 30], magnitude=[4, 5]`)

One more thing, some baseline images are too big because you're plotting too many beachballs (300KB + 500 KB).

Reply:
These are all good points, @seisman. We will re-think the tests — we can likely do away with files. I think more tests like  [`test_meca_spec_dictionary()`](https://github.com/tjnewton/pygmt/blob/45b01de8c68dce8103faaa0dc3b98e2b71347ec0/pygmt/tests/test_meca.py#L80), but maybe with different mechanism types, are what we need.

> * pass multiple focal mechanisms like dict(strike=[0, 30], dip=[30, 60], rake=[30, 30], magnitude=[4, 5])

I haven't tested, but I don't think this works yet.